src/hooks/useCart/useCart.test.tsx

```diff
@@ -301,7 +319,36 @@ describe("useCart", () => {
         }
       ]);
     });
+    it("should have cart state be NO_QUOTA when negative quota is received", async () => {
+      expect.assertions(3);
+      mockGetQuota.mockReturnValueOnce(mockQuotaResSingleIdNegativeQuota);
 
+      const ids = ["ID1"];
+      const { result, waitForNextUpdate } = renderHook(
+        () => useCart(ids, key, endpoint),
+        { wrapper: Wrapper }
+      );
+      expect(result.current.cartState).toBe("FETCHING_QUOTA");
+
+      await waitForNextUpdate();
+      expect(result.current.cartState).toBe("NO_QUOTA");
+      expect(result.current.cart).toStrictEqual([
+        {
+          category: "toilet-paper",
+          identifierInputs: [],
+          lastTransactionTime: transactionTime,
+          maxQuantity: -1,
+          quantity: -1
```

Comment:
Although it's not used once the app reaches this state, the cart should be more resilient by not allowing negative values. So if for whatever reason the cart is seen, the user will never see negative values.